### PR TITLE
[collector] Use ottl in filterprocessor example

### PR DESCRIPTION
**Pull request by TylerHelmuth**
This PR updates the filterprocessor example to use the preferred OTTL syntax instead of the deprecated include/exclude interface.

Related to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18642.

### CODE REVIEW

File: content/en/docs/collector/transforming-telemetry.md [hunk 1]
```diff
@@ -23,38 +23,35 @@ a significant impact on collector performance.
...
+      span:
+        - |
+        resource.attributes["service.name"] != "app1" and
+        resource.attributes["service.name"] != "app2" and
+        resource.attributes["service.name"] != "app2"
 ```
 
-To only block spans from a service called development while allowing all other
-spans, an exclude rule is used:
+To only drop spans from a service called development while keeping all other
```
Comment (codeboten) [inline content/en/docs/collector/transforming-telemetry.md:45]:
```suggestion
To only drop spans from a service called `service1` while keeping all other
```

File: content/en/docs/collector/transforming-telemetry.md [hunk 2]
```diff
@@ -23,38 +23,35 @@ a significant impact on collector performance.
...
-    spans:
-      exclude:
-        match_type: strict
-        services:
-          - development
+  filter/ottl:
+    error_mode: ignore
+    traces:
+      span:
+        - resource.attributes["service.name"] == "development"
```
Comment (codeboten) [inline content/en/docs/collector/transforming-telemetry.md:54]:
```suggestion
        - resource.attributes["service.name"] == "service1"
```

**Review by codeboten — APPROVED** [review]

File: content/en/docs/collector/transforming-telemetry.md [hunk 3]
```diff
@@ -23,38 +23,35 @@ a significant impact on collector performance.
 **Processor**:
 [filter processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/filterprocessor)
 
-The filter processor allows users to filter telemetry based on `include` or
-`exclude` rules. Include rules are used for defining "allow lists" where
-anything that does _not_ match include rules is dropped from the collector.
-Exclude rules are used for defining "deny lists" where telemetry that matches
-rules is dropped from the collector.
+The filter processor allows users to filter telemetry using
+[OTTL](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/ottl/README.md).
```
Comment (cartermp) [inline content/en/docs/collector/transforming-telemetry.md:27]: You'll want to add `OTTL` to the cspell section at the top of the doc

**Review by cartermp — APPROVED** [review]
LGTM modulo the small suggestions

File: content/en/docs/collector/transforming-telemetry.md (hunk 3 above)
Comment (svrnm) [reply content/en/docs/collector/transforming-telemetry.md:27]: don't we have OTTL in cspell yet? 😮

**Review by theletterf — APPROVED** [review]

**Pull request merged**